<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>E! Data Review 10.06.2019</title>
  <link rel="stylesheet" href="https://stackedit.io/style.css" />
</head>

<body class="stackedit">
  <div class="stackedit__left">
    <div class="stackedit__toc">
      
<ul>
<li><a href="#gene-expression-data-for-wormbase-parasite">Gene expression data for WormBase ParaSite</a>
<ul>
<li><a href="#worms-and-wormbase-parasite">Worms and WormBase ParaSite</a></li>
<li><a href="#about-gene-expression-data">About gene expression data</a></li>
<li><a href="#our-solution">Our solution</a></li>
<li><a href="#details">Details</a></li>
<li><a href="#future-directions">Future directions</a></li>
<li><a href="#end">End</a></li>
</ul>
</li>
</ul>

    </div>
  </div>
  <div class="stackedit__right">
    <div class="stackedit__html">
      <h1 id="gene-expression-data-for-wormbase-parasite">Gene expression data for WormBase ParaSite</h1>
<p><img src="https://parasite.wormbase.org/i/gallery/gene_exp.png" alt="Gene expression studies logo"><br>
Written for Ensembl Data Review, 10.06.2019</p>
<h2 id="worms-and-wormbase-parasite">Worms and WormBase ParaSite</h2>
<h3 id="about-parasitic-worms">About parasitic worms</h3>
<p>Some organisms have adapted to live some of their lives inside human hosts. Here’s a drawing of a pair of schistosomes living in a blood vessel:<br>
<img src="https://www.cdc.gov/dpdx/schistosomiasis/images/7/S_mansoni_adult_Lammie2.jpg" alt="Adult schistosomes"><br>
Estimated 252 million people have these.</p>
<p>Additionally, 1.5 billion have soil-transmitted helminths, 120 million have lymphatic filariasis, and 37 million have river blindness. (figures from Wikipedia / World Health Organization).</p>
<p>Genomic methods offer insights into the biology of these animals, which could translate into potential new treatments and drug targets.</p>
<h3 id="about-us">About us</h3>
<p>WormBase ParaSite is an EBI - Sanger collaboration, and we run a resource with genomes of parasitic worms using Ensembl software. We also host data about other worms : plant pests or free-living worms relevant to basic research.</p>
<p>I’m a bioinformatician on the project, running production and doing small scale software development.</p>
<p>We rely on Ensembl software and pipelines for most of our operations. We also have odd bits and customizations for the kind of data that is available for worms.</p>
<p>This talk is about how we added gene expression data to the site - one of these custom bits. No EBI resource offered the functionality we needed, so we embarked on approximately half a year of design and development to make it available.</p>
<h2 id="about-gene-expression-data">About gene expression data</h2>
<p>Apart from DNA, you can also sequence RNA. Not all genes are transcribed at the same time, so the results will vary based on what is going on in the cells.</p>
<p>You can try measure “normal” state of cells e.g. as the worms develop, or measure response to a treatment e.g. with a potential drug.</p>
<h3 id="use-cases-of-gene-expression-data">Use cases of gene expression data</h3>
<ul>
<li>Confirming gene models</li>
<li>Getting hints at gene function</li>
<li>Validating original analyses done by the data set’s authors</li>
<li>Querying based on expression profile</li>
</ul>
<h3 id="our-need-for-a-custom-solution-instead-of-the-usual-expression-atlas-widget">Our need for a custom solution instead of the usual Expression Atlas widget</h3>
<p>Data mostly deposited to ENA, which Atlas doesn’t curate.</p>
<p>We have lower standards: sequencing parasites is difficult and there isn’t much funding for it, so we need to be more inclusive in curation. The data is worse as a result, so we also need to be more conservative in analysis.</p>
<h3 id="rnaseq-er-as-source-of-data">RNASeq-er as source of data</h3>
<h4 id="introduction">Introduction</h4>
<p>Ran by EBI Functional Genomics Team, same team that runs Atlas.</p>
<p>They have data for (almost) all eukaryotic genomes from Ensembl, Ensembl Genomes, and WormBase ParaSite.</p>
<h4 id="overview">Overview</h4>
<p><img src="https://i.imgur.com/YNBl69U.jpg" alt="RNASeq-er about"></p>
<h4 id="what-rnaseq-er-doesnt-offer">What RNASeq-er doesn’t offer</h4>
<ul>
<li>Aggregate or compare data - only alignments and quantifications per run</li>
<li>Serve alignment tracks through a genome browser</li>
<li>Data per worm gene</li>
<li>Metadata about studies</li>
</ul>
<h2 id="our-solution">Our solution</h2>
<p>Here is what we provide to our users:</p>
<h3 id="alignment-tracks">Alignment tracks</h3>
<p>Aggregated into a UCSC-style track hub, and added to our genome browser. We also offer JBrowse as an alternative genome browser, and the tracks work much better there!</p>
<iframe src="https://parasite.wormbase.org/jbrowse/index.html?data=%2Fjbrowse-data%2Ffasciola_hepatica_prjeb25283%2Fdata&amp;loc=scaffold10x_213_pilon%3A1005821..1025020&amp;tracks=DNA%2CGene_Models%2CRNASeq%2FERR576953%2CRNASeq%2FERR576962&amp;highlight=" width="100%" height="600px">&#10;</iframe><h3 id="studies-page">Studies page</h3>
<p>List of all studies for a species, with metadata - how runs group into conditions,  description of what was done in the study, any publications</p>
<iframe src="https://parasite.wormbase.org/expression/schistosoma_mansoni_prjea36577/index.html" width="100%" height="600px">&#10;</iframe><h3 id="gene-page">Gene page</h3>
<p>Implemented as a list of custom tabs on the gene page.</p>
<iframe src="https://parasite.wormbase.org/Schistosoma_mansoni_prjea36577/Gene/WBPSExpressionLife_cycle?g=Smp_316150;r=SM_V7_2:9412952-9413667;t=Smp_316150.1" width="100%" height="600px">&#10;</iframe><h2 id="details">Details</h2>
<h3 id="data-we-offer">Data we offer</h3>
<ul>
<li>read counts and transcripts per million reads (TPMs) for each gene in each ENA run</li>
<li>TPMs in a condition – median value for all runs representing the same condition</li>
<li>differential expression analysis between selected pairs of conditions</li>
</ul>
<h3 id="metadata">Metadata</h3>
<p>There are 584 studies available from RNASeq-er for our worms. I have selected and curated metadata for 125/279 of non-C. elegans studies, and 29 out of 305 for C. elegans studies.</p>
<p>The curation is semi-automatic: we retrieve data for all the studies out there, apply heuristics, and then run extensive checks. Results are stored in files and managed with  the  source code through <code>git</code>.</p>
<h4 id="example">Example</h4>
<p>This is a study for H. glycines, a pest of soybean. for the following publication:</p>
<p><em>Novel global effector mining from the transcriptome of early life stages of the soybean cyst nematode Heterodera glycines</em> (Gardner et al, 2018)</p>
<p>The authors deposited data to ENA with the following annotation:</p>

<table>
<thead>
<tr>
<th>Run</th>
<th>Condition</th>
<th>sample_name</th>
<th>host</th>
<th>geographic_location</th>
<th>isolate</th>
<th>isolation_source</th>
</tr>
</thead>
<tbody>
<tr>
<td>SRR6230579</td>
<td>ppJ2 PA3 3</td>
<td>ppJ2 PA3 3</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230580</td>
<td>pJ2 PA3-IC 1</td>
<td>pJ2 PA3-IC 1</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230581</td>
<td>ppJ2 PA3 1</td>
<td>ppJ2 PA3 1</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230582</td>
<td>ppJ2 PA3 2</td>
<td>ppJ2 PA3 2</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230583</td>
<td>pJ2 PA3-C 1</td>
<td>pJ2 PA3-C 1</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230584</td>
<td>pJ2 PA3-C 2</td>
<td>pJ2 PA3-C 2</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230585</td>
<td>pJ2 PA3-IC 2</td>
<td>pJ2 PA3-IC 2</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230586</td>
<td>pJ2 PA3-IC 3</td>
<td>pJ2 PA3-IC 3</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230587</td>
<td>pJ2 PA3-C 3</td>
<td>pJ2 PA3-C 3</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
</tbody>
</table><p>I have read the paper, and found out that the data is about the pest invading the plant, before and after invasion.</p>

<table>
<thead>
<tr>
<th>Run</th>
<th>Condition</th>
<th>developmental_stage</th>
<th>host_compatibility</th>
<th>host</th>
<th>geographic_location</th>
<th>isolate</th>
<th>isolation_source</th>
</tr>
</thead>
<tbody>
<tr>
<td>SRR6230581</td>
<td>pre-parasitic juvenile</td>
<td>pre-parasitic second-stage juvenile</td>
<td></td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230582</td>
<td>pre-parasitic juvenile</td>
<td>pre-parasitic second-stage juvenile</td>
<td></td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230579</td>
<td>pre-parasitic juvenile</td>
<td>pre-parasitic second-stage juvenile</td>
<td></td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230583</td>
<td>parasitic juvenile, compatible host</td>
<td>parasitic second-stage juvenile</td>
<td>compatible host</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230584</td>
<td>parasitic juvenile, compatible host</td>
<td>parasitic second-stage juvenile</td>
<td>compatible host</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230587</td>
<td>parasitic juvenile, compatible host</td>
<td>parasitic second-stage juvenile</td>
<td>compatible host</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230580</td>
<td>parasitic juvenile, resistant host</td>
<td>parasitic second-stage juvenile</td>
<td>resistant host</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230585</td>
<td>parasitic juvenile, resistant host</td>
<td>parasitic second-stage juvenile</td>
<td>resistant host</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
<tr>
<td>SRR6230586</td>
<td>parasitic juvenile, resistant host</td>
<td>parasitic second-stage juvenile</td>
<td>resistant host</td>
<td>Glycine max</td>
<td>USA</td>
<td>PA3</td>
<td>greenhouse population</td>
</tr>
</tbody>
</table><p>Then, the analysis pipeline compared:</p>
<ul>
<li><code>pre-parasitic juvenile vs parasitic juvenile, compatible host</code></li>
<li><code>pre-parasitic juvenile vs parasitic juvenile, resistant host</code></li>
<li><code>parasitic juvenile: compatible vs resistant host</code></li>
</ul>
<h3 id="software">Software</h3>
<p>All written by me and Tuan in very nice Perl.</p>
<p><img src="https://cdn.dribbble.com/users/1047458/screenshots/2991320/dribbble_perlchainsaw.png" alt="Perl - swiss army chainsaw"></p>
<p>We relied on simple techniques:</p>
<ul>
<li>metadata is curated as files, and stored with source code</li>
<li>the studies page is a single HTML document per species</li>
<li>results are in TSV files, easy to download from the studies page</li>
<li>data for the gene page is constructed by the web server running <code>grep "$gene_id"</code> on TSV files</li>
</ul>
<p>We didn’t need:</p>
<ul>
<li>databases</li>
<li>Hive: there’s one entry point to the whole program</li>
<li>JavaScript</li>
</ul>
<h2 id="future-directions">Future directions</h2>
<p>We can still make it better, but the returns will be diminishing:<br>
<img src="http://cdn.economicsdiscussion.net/wp-content/uploads/2015/10/clip_image00210.jpg" alt="diminishing returns"></p>
<p>We will keep the dataset updated as more data comes out - estimating 10 studies per release, so it’s not much work to keep it up to date.</p>
<p>I am planning incremental improvements: more informative studies page, differential expression data in a better format, etc.</p>
<p>I’d expand the project outside WormBase ParaSite, if I had a sustainable plan for it!</p>
<h2 id="end">End</h2>
<h3 id="acknowledgements">Acknowledgements</h3>

<table style="text-align: center;">
<thead>
<tr>
<th>Who I thank and what for</th>
<th>Inspiring the project</th>
<th>Development</th>
<th>Feedback and input</th>
<th>Running RNASeq-er</th>
<th>Letting my job exist</th>
</tr>
</thead>
<tbody>
<tr>
<td>Wojtek Bażant (me, WBPS bioinformatician)</td>
<td></td>
<td>✅</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>Tuan Le (WBPS web developer)</td>
<td></td>
<td>✅</td>
<td>✅</td>
<td></td>
<td></td>
</tr>
<tr>
<td>Faye Rodgers (WBPS bioinformatician)</td>
<td></td>
<td></td>
<td>✅</td>
<td></td>
<td></td>
</tr>
<tr>
<td>Kevin Howe (WBPS co-PI, EBI)</td>
<td></td>
<td></td>
<td>✅</td>
<td></td>
<td>✅</td>
</tr>
<tr>
<td>Matt Berriman (WBPS co-PI, Sanger)</td>
<td>✅</td>
<td></td>
<td>✅</td>
<td></td>
<td>✅</td>
</tr>
<tr>
<td>WormBase ParaSite users</td>
<td>✅</td>
<td></td>
<td></td>
<td></td>
<td>✅</td>
</tr>
<tr>
<td>Jonathan Manning (EBI FG team)</td>
<td></td>
<td></td>
<td></td>
<td>✅</td>
<td></td>
</tr>
<tr>
<td>Suhaib Mohammed (EBI FG team)</td>
<td></td>
<td></td>
<td></td>
<td>✅</td>
<td></td>
</tr>
<tr>
<td>BBSRC, our funding body</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>✅</td>
</tr>
</tbody>
</table><h3 id="thank-you-for-listening">Thank you for listening</h3>

<table>
<thead>
<tr>
<th>Who I thank and what for</th>
<th>Listening</th>
</tr>
</thead>
<tbody>
<tr>
<td>You</td>
<td>✅</td>
</tr>
</tbody>
</table><p>Please ask me questions!</p>
<p><img src="https://pixel.nymag.com/imgs/daily/grub/2017/05/12/12-sushi.w700.h700.jpg" alt="sushi instead of worm picture"></p>

    </div>
  </div>
</body>

</html>
