<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>How I&#x27;ve failed WBPS14</title>
  <link rel="stylesheet" href="https://stackedit.io/style.css" />
</head>

<body class="stackedit">
  <div class="stackedit__left">
    <div class="stackedit__toc">
      
<ul>
<li><a href="#how-ive-failed---wbps14-edition">How I’ve failed - WBPS14 edition</a>
<ul>
<li><a href="#wbps13-bugs">WBPS13 bugs</a></li>
<li><a href="#lost-multiple-days-of-production">Lost multiple days of production</a></li>
<li><a href="#near-misses">Near misses</a></li>
<li><a href="#other">Other</a></li>
</ul>
</li>
</ul>

    </div>
  </div>
  <div class="stackedit__right">
    <div class="stackedit__html">
      <h1 id="how-ive-failed---wbps14-edition">How I’ve failed - WBPS14 edition</h1>
<h2 id="wbps13-bugs">WBPS13 bugs</h2>
<h3 id="no-paralogs-and-orthologs-for-wormbase-core-species-on-ftp-and-also-m.-japonica">No paralogs and orthologs for WormBase core species on FTP, and also M. japonica</h3>
<p>Wojtek noticed after writing an FTP check that he thought would be good to have, and the check not succeeding.</p>
<h3 id="clash-in-biomart-of-m.-japonica-and-p.-japonicus-prjeb27334">Clash in BioMart of M. japonica and P. japonicus PRJEB27334</h3>
<p>Results in P. japonicus not in BioMart UI, M. japonica “twice”, and P. japonicus orthologs appearing when asking for M. japonica. You can query for M. japonica genes okay.</p>
<p>Genomes in the merged mart need to have a short name to be used as key, or else stuff won’t work. An attempt to resolve this is a Golden Species Regex which takes a BioMart name and produces a short name, carefully written and then copy-pasted around for WBPS11.</p>
<p>Before GSR we had Bruce’s Slightly Worse Regex, that clashed on species of Trichinella and more stuff, and a correspondingly inadequate BioMart. The GSR resolves the clashes of BSWR, but still clashes on two WBPS13 genomes, there was no check that GSR doesn’t clash, and it’s not implemented in one place.</p>
<p>Wojtek noticed when looking at the issue above.</p>
<h3 id="wbps13-and-before-bad-species-tree-in-biomart">WBPS13 and before: Bad species tree in BioMart</h3>
<p>The script that dumps it, <code>dump_taxon_tree.pl</code>, didn’t know about the GSR and still used BSWR.<br>
Wojtek noticed when reading the source code of that program for an unrelated reason.</p>
<h3 id="did-not-notice-that-the-custom-track-hubs-are-gone">did not notice that the custom track hubs are gone</h3>
<p>For RNASeq, etc. No checklist involves confirming they exist.</p>
<h3 id="showed-bad-stuff-on-emu-archive-page">showed bad stuff on emu archive page</h3>
<p>This is a bug: <a href="https://github.com/wbazant/wormbase-pipeline/commit/4c67232f54a">capture groups get messed up</a><br>
The bug was limited to one species, <em>E. multilocularis</em>.</p>
<h3 id="no-repeats-for-two-genomes">No repeats for two genomes</h3>
<p>I have found that panagrellus_redivivus_prjna186477 and caenorhabditis_tropicalis_prjna53597 are missing repeats. It’s been like that for a while, probably like five releases, and it doesn’t matter. I guess the real failure was not having a check.</p>
<h3 id="refseq-as-a-query-filter">RefSeq as a query filter</h3>
<p>There’s <code>RefSeq_mRNA</code>  (for C. elegans, where we know the xrefs) and <code>RefSeq_mRNA__predicted</code> (for everything else, where we try to reconcile them) and also a bug that made me redo things manually. I’ve probably noticed and probably merged them in the _dm table, but not in the boolean column.</p>
<h2 id="lost-multiple-days-of-production">Lost multiple days of production</h2>
<h3 id="deleted-a-merging-script">Deleted a merging script</h3>
<p>As part of cleanup of the <code>eg-biomart</code> repository I’ve deleted <code>merge-new.pl</code>, which I’ve spent a week over in WBPS13, and did not commit. Notably, <code>merge.pl</code> did not work, so I got to have another go at merging Biomarts and wrote <code>create_merging_sql.pl</code>.</p>
<p>Possibly I did a better job: <code>merge-new.pl</code> tried to be idempotent which wasn’t actually that helpful because the script is more complicated - benefit being, that we could redo parts of merging.<br>
<code>create_merging_sql.pl</code> creates all the SQL, and then if we want to manually rerun something we can. Not done yet.</p>
<h2 id="near-misses">Near misses</h2>
<h3 id="forgot-a-step-about-examples-and-descriptions">Forgot a step about examples and descriptions</h3>
<p><a href="https://www.ebi.ac.uk/seqdb/confluence/display/WORMBASE/Examples+and+descriptions">https://www.ebi.ac.uk/seqdb/confluence/display/WORMBASE/Examples+and+descriptions</a></p>
<h2 id="other">Other</h2>
<h3 id="eg-pipelines-was-old">eg-pipelines was old</h3>
<p>I’ve switched it midway of redoing missed DNAFeatures, and it resulted in a<br>
mixed run for caenorhabditis_tropicalis_prjna53597 and having to update instructions for panagrellus_redivivus_prjna186477. I’ve added --production_db ensembl_production_parasite.</p>

    </div>
  </div>
</body>

</html>
