<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>MBio on EDA as prototype for systems biology</title>
  <link rel="stylesheet" href="https://stackedit.io/style.css" />
</head>

<body class="stackedit">
  <div class="stackedit__left">
    <div class="stackedit__toc">
      
<ul>
<li><a href="#mbio-on-eda-as-a-prototype-for-systems-biology">MBio on EDA as a prototype for systems biology</a>
<ul>
<li><a href="#currently">Currently</a></li>
<li><a href="#new-data-type-for-mbio---metabolomics">New data type for MBio - metabolomics</a></li>
<li><a href="#mbio-on-eda">MBio on EDA</a></li>
<li><a href="#a-prototype-for-systems-biology-">A prototype for systems biology ?</a></li>
</ul>
</li>
</ul>

    </div>
  </div>
  <div class="stackedit__right">
    <div class="stackedit__html">
      <h1 id="mbio-on-eda-as-a-prototype-for-systems-biology">MBio on EDA as a prototype for systems biology</h1>
<h2 id="currently">Currently</h2>
<p>MicrobiomeDB is implemented as a WDK site, and lets users look at sample annotations together with their taxonomic abundance. There’s an EDA prototype, with more data.</p>
<h3 id="what-can-users-learn-from-microbiomedb-now">What can users learn from MicrobiomeDB now?</h3>
<p>Some general patterns show up in multiple studies:</p>
<ul>
<li>newborns have simple microbiomes with a few common species</li>
<li>as infants get older, their gut microbiome becomes more diverse</li>
<li>body sites can be distinguished by their bacteria populations</li>
</ul>
<p>Individual data sets also contain more specific insights. You can usually verify what the publication has shown, as well as look at the context of it.</p>
<p>Example: cerebrospinal fluid from patients in Uganda with two forms of hydrocephalus - useful if you want to know what lives in CSF, and also, the publication found a specific bug that causes one of the forms of hydrocephalus.</p>
<h3 id="comparison-of-wdk-models">Comparison of WDK models</h3>

<table>
<thead>
<tr>
<th></th>
<th>MicrobiomeDB</th>
<th>PlasmoDB</th>
</tr>
</thead>
<tbody>
<tr>
<td>record plurality</td>
<td>one - MicrobiomeSampleRecord</td>
<td>many - genes, pathways, compounds …</td>
</tr>
<tr>
<td>records per study?</td>
<td>✓</td>
<td>✗</td>
</tr>
<tr>
<td>user focuses on</td>
<td>groups of records together</td>
<td>individual records or their properties</td>
</tr>
<tr>
<td>core features</td>
<td>“Analyze results”</td>
<td>search, result lists, record pages</td>
</tr>
<tr>
<td>“Analyze results”</td>
<td>shows patterns and relations</td>
<td>extracts details from lists - e.g. gene set enrichment</td>
</tr>
<tr>
<td>information density</td>
<td>low</td>
<td>high</td>
</tr>
<tr>
<td>experimental designs</td>
<td><em>sort of</em></td>
<td>yes - e.g. test vs ref in triplicate</td>
</tr>
</tbody>
</table><h2 id="new-data-type-for-mbio---metabolomics">New data type for MBio - metabolomics</h2>
<p>We currently have an unintegrated data type - functional profiling - and we also started to talk about adding metabolomics data.</p>
<h3 id="metabolite-data-in-genomic-sites">Metabolite data in genomic sites</h3>
<p><a href="https://plasmodb.org/plasmo/app/search/compound/CompoundsByFoldChange">PlasmoDB - compounds by fold change search</a></p>
<p>This is data about amounts of compounds, measured in a number of experimental conditions related to a Plasmodium infection.</p>
<p>Using differential abundance, the user can find compounds that have something to do with the experimental condition. Compounds are a record type, so each result in a differential abundance test can be investigated - maybe clicked on and read about.</p>
<h5 id="would-this-work-for-mbio">Would this work for MBio?</h5>
<p><em>… probably not.</em></p>
<ol>
<li>PlasmoDB takes a relation between compounds and genes as a fact - so you can transform the list of compounds into a list of genes.<br>
On the other hand, it’s not known how compounds relate to something on MicrobiomeDB - e.g. bacterial populations - and it’s instead one of the things to be learned.</li>
<li>In PlasmoDB, searching compounds by fold change leads to lists of results that might contain something potentially interesting. In MicrobiomeDB, there’s not much point in looking at record pages - they aren’t really interesting or relevant for anything.</li>
<li>In PlasmoDB the shown experiment has a cleanly defined design - regular vs. infected red blood cells, at three different pHs. The MicrobiomeDB studies have more samples with a larger amount of variables.</li>
</ol>
<p>There’s more hope in the new stuff!</p>
<h2 id="mbio-on-eda">MBio on EDA</h2>
<h3 id="demo---entity-graph">Demo - entity graph</h3>
<p><a href="https://feature.clinepidb.org/ce.feature/app/eda/DS_22e3c3c070">https://feature.clinepidb.org/ce.feature/app/eda/DS_22e3c3c070</a></p>
<p>Already there:</p>
<ul>
<li>shape of the entity graph (host -&gt; sample -&gt; WGS and amplicon assay)</li>
<li>variables for taxa and functional profiles</li>
<li>EDA visualizations like scatter plot - almost there</li>
</ul>
<p>The main thing missing is the ability to look at many variables together. I complained and got the plans for:</p>
<h3 id="the-compute-service">The compute service</h3>
<p><img src="https://i.imgur.com/k6DV3yB.png" alt="diagram"><br>
Subsetting, merging, computing, merging again, data. Ellie will adapt the blast queue system. Details not included.</p>
<h4 id="or">or</h4>
<p>I’m finding it more easily imaginable to add the Shiny apps back in:<br>
<img src="https://i.imgur.com/vHOLm5u.png" alt="shiny in"><br>
The current Shiny apps read tab files written somewhere temporary, know where to look for them given a URL, and are integrated in the display as an iframe, so it feels like less of a stretch.</p>
<p>They were designed for taxon abundance, but maybe there’s some generality to them. For example, if a user were to put the functional profiling data into the relative abundance app: most of the functions are unknown, so there is no equivalent to the most abundant taxa. On the other hand, sorting pathways by variance could be very interesting - a missing pathway between different environments like body sites might mean an adaptation to that environment. BioBakery even has a tool to look for strains of the same bacteria that have a pathway missing between samples but sorting by variance would also bring them up.</p>
<h3 id="new-visualizations">New visualizations?</h3>
<p>ClinEpiDB has its visualisations for looking at pairs of variables. For example, a histogram takes in two numeric variables, and shows the correlation or trend between them. VectorBase needs those, and a map for geospatial data.<br>
For MicrobiomeDB, Ann + Danielle + Dan mostly worked on designing a better web version of the current apps.</p>
<p>I think a long-term way forward is through visualisations that work for our data, which are <em>lists of numbers with labels</em>.</p>
<p>They show up a bit in ClinEpiDB too, as multi-filters - although these are usually lists of booleans with labels.</p>
<h4 id="example-correlation-heatmap">Example: correlation heatmap</h4>
<p>Correlation heatmaps:</p>
<ul>
<li>show relationships between two key-to-numeric-value variables</li>
<li>don’t show individual samples</li>
</ul>
<p><a href="%5Bhttps://www.frontiersin.org/articles/10.3389/fgene.2019.00454/full%5D(https://www.frontiersin.org/articles/10.3389/fgene.2019.00454/full)">Tang et al. (2019)</a>  used metagenomic sequencing and metabolomics to study influence of diet.</p>
<p>They looked at three variables: nutrient intakes, taxon abundances, and amounts of measured metabolite. Two are currently supported on our site, one is new. A generic correlation heatmap would do it all!</p>
<h5 id="nutrient-intake-vs.-taxon-abundance">Nutrient intake vs. taxon abundance</h5>
<p><img src="https://www.frontiersin.org/files/Articles/451177/fgene-10-00454-HTML/image_m/fgene-10-00454-g003.jpg" alt=" taxon abundance and caloric intake of a nutrient computed from a questionnaire"></p>
<h5 id="taxon-abundance-vs.-metabolite-level">Taxon abundance vs. metabolite level</h5>
<p><img src="https://www.frontiersin.org/files/Articles/451177/fgene-10-00454-HTML/image_m/fgene-10-00454-g004.jpg" alt="taxon abundance and measured metabolite"></p>
<h2 id="a-prototype-for-systems-biology-">A prototype for systems biology ?</h2>
<blockquote>
<p><strong>systems biology</strong> is a field of research which focuses on understanding entire biological systems  in contrast to the focus on single genes or proteins.</p>
</blockquote>
<p>There is value to taking the reductionist view of biology the WDK is good for, but there are other approaches now - example: <a href="https://www.sanger.ac.uk/science/tools/mca/mca/">Malaria Cell Atlas</a> .</p>
<p>WDK is not a good fit for MicrobiomeDB, because in metagenomics, individual samples aren’t really important - a bit like a single household or individual isn’t of interest in epidemiological data.</p>
<p>More data types, more visualizations, and more putting stuff on EDA is clearly going to be the future of MicrobiomeDB, but maybe also of the whole portal!</p>

    </div>
  </div>
</body>

</html>
