<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>ParaSite 11</title>
  <link rel="stylesheet" href="https://stackedit.io/style.css" />
</head>

<body class="stackedit">
  <div class="stackedit__html"><h1 id="lab-meeting-16.10.2018">Lab meeting 16.10.2018</h1>
<h2 id="after-wbps11">After WBPS11</h2>
<h3 id="updates-of-popular-genomes">Updates of popular genomes</h3>
<p><em>S. mansoni, F. hepatica, H. contortus</em><br>
Large release, it turned out to be quite disruptive. In response:</p>
<ul>
<li>S. mansoni mapping made available as cross-references</li>
<li>Blog post telling people how they can cope</li>
</ul>
<h3 id="comments-in-parasite">Comments in ParaSite</h3>
<p>As of a week ago there were three comments on the live site, by:</p>
<ul>
<li>Tuan (for test, and deleted)</li>
<li>me (later deleted)</li>
<li>Alan (our best comment)</li>
</ul>
<h3 id="rnaseq-data">RNASeq data</h3>
<p>I have little data on usage. The people I demonstrated the feature to in our lab seem to interact with it via Ensembl genome browser. I designed it for the JBrowse browser, and support elsewhere was added later, as a bit of a bolt-on.</p>
<h2 id="wbps12">WBPS12</h2>
<h3 id="genomes">Genomes</h3>
<ul>
<li>Acrobeloides nanus, a clade IV nematode</li>
<li>Ancylostoma ceylanicum: new annotation</li>
<li><strong>Hymenolepis microstoma</strong></li>
<li>Meloidogyne arenaria, alternative assembly</li>
<li>Meloidogyne graminicola, draft assembly</li>
<li><strong>Taenia multiceps</strong></li>
<li>S. mansoni annotation update</li>
</ul>
<ul>
<li>WormBase core species up to version WS267.</li>
</ul>
<h3 id="better-c.-elegans-references">Better C. elegans references</h3>
<p>I spent ~three weeks on the xref pipeline to incorporate a C. elegans protein mapping from WormBase.<br>
Result: more accurate and complete UniProt, RefSeq mRNA, RefSeq protein entries.</p>
<h3 id="archiving-ids">Archiving IDs</h3>
<p>I’ve adapted Ensembl’s ID mapping pipeline, based on exon-on-exon matching with exonerate that gets propagated to transcript and gene level.</p>
<h4 id="mapping-success-rate">Mapping success rate</h4>

<table>
<thead>
<tr>
<th>Genome</th>
<th>number of genes in current assembly</th>
<th>release of WBPS with previous assembly</th>
<th>genes successfully mapped</th>
<th>genes in previous assembly</th>
<th>fraction successfully mapped</th>
</tr>
</thead>
<tbody>
<tr>
<td>ancylostoma_ceylanicum_prjna72583</td>
<td>11783</td>
<td>WBPS11</td>
<td>7564</td>
<td>15892</td>
<td>0.476</td>
</tr>
<tr>
<td>ascaris_suum_prjna62057</td>
<td>17974</td>
<td>WBPS9</td>
<td>9468</td>
<td>15260</td>
<td>0.620</td>
</tr>
<tr>
<td>fasciola_hepatica_prjeb25283</td>
<td>16806</td>
<td>WBPS10</td>
<td>7564</td>
<td>22676</td>
<td>0.334</td>
</tr>
<tr>
<td>haemonchus_contortus_prjeb506</td>
<td>19430</td>
<td>WBPS10</td>
<td>11439</td>
<td>21869</td>
<td>0.523</td>
</tr>
<tr>
<td>meloidogyne_incognita_prjeb8714</td>
<td>45351</td>
<td>WBPS10</td>
<td>11977</td>
<td>19212</td>
<td>0.623</td>
</tr>
</tbody>
</table><p>This is already 5 to 20% more than running the pipeline with default parameter values.</p>
<h4 id="handling-unmapped-genes">Handling unmapped genes</h4>
<p>Unmapped genes get killed, and:</p>
<ul>
<li>user can search for them, and see what happened (i.e. they got killed)</li>
<li>user can get the protein sequence</li>
</ul>
<h2 id="rnaseq-data-expression-querying">RNASeq data: Expression querying</h2>
<h3 id="where-we-are">Where we are</h3>
<p>Lots of S. mansoni data from our lab published and available in ENA<br>
Comprehensive treatment of all RNASeq data sets in ENA for our species: currently over 10k runs with metadata<br>
No quantitative results - we show alignments in genome browser</p>
<h3 id="goal-selective-treatment-and-deeper-integration-of-some-data-sets">Goal: selective treatment and deeper integration of some data sets</h3>
<ul>
<li>profiling baseline expression across sexes, life stages, and organism parts</li>
<li>differential expression across contrasts</li>
<li>basically informative data - gene has expression evidence or not - when this is all we can tell</li>
<li>aggregated information on gene pages</li>
<li>query interface</li>
</ul>
<h3 id="required-effort">Required effort</h3>
<ul>
<li>shortlist valuable experiments</li>
<li>schema for metadata</li>
<li>curate data in that format</li>
<li>develop analyses (wrappers around existing tools like DESeq2)</li>
<li>pipelines to run above</li>
<li>design database schema</li>
<li>store analysis results in the database</li>
<li>query code for gene pages</li>
<li>UI design / implementation for gene pages</li>
<li><em>integrating queries - the hard part!</em></li>
</ul>
<h3 id="querying">Querying</h3>
<p>We could use BioMart for purposes of integrating the data: it’s limited and hard to work with, but it’s our best bulk query tool.</p>
<h4 id="filters-and-attributes">Filters and attributes</h4>
<p>Attributes: TPM and standard deviation for each group of samples, or fold change and p-value per contrast for differential expression.</p>
<p>Filters are hard - I would have to split above values into ranges and add a filter for each group of samples or contrast, but it doesn’t scale well.</p>
</div>
</body>

</html>
