<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>short read metagenomics spec</title>
  <link rel="stylesheet" href="https://stackedit.io/style.css" />
</head>

<body class="stackedit">
  <div class="stackedit__left">
    <div class="stackedit__toc">
      
<ul>
<li><a href="#short-read-metagenomics-in-microbiomedb">Short read metagenomics in MicrobiomeDB</a>
<ul>
<li><a href="#summary-of-deliverables">Summary of deliverables</a></li>
<li><a href="#workflow">Workflow</a></li>
<li><a href="#data-model">Data model</a></li>
<li><a href="#sample-record-page">Sample record page</a></li>
<li><a href="#searches">Searches</a></li>
<li><a href="#analysis-tools">Analysis tools</a></li>
<li><a href="#future-directions">Future directions</a></li>
</ul>
</li>
</ul>

    </div>
  </div>
  <div class="stackedit__right">
    <div class="stackedit__html">
      <h1 id="short-read-metagenomics-in-microbiomedb">Short read metagenomics in MicrobiomeDB</h1>
<h2 id="summary-of-deliverables">Summary of deliverables</h2>
<ol>
<li>New workflow using methods that map to known references and produce all data required</li>
<li>Sample pages show taxon counts comparable with 16s counts</li>
<li>Gene and pathway data are the presented as the main way to use short read metagenomic data on the site</li>
<li>The data model is adapted to the new data through changing current and adding new attributes to MicrobiomeSampleRecord</li>
<li>Sample details search is modified to allow selecting only samples with short reads</li>
<li>New pathway abundance search is visible for studies that have short read metagenomic data</li>
<li>New gene / pathway abundance app is created similar to the relative abundance app, OR the relative abundance is extended to display gene and pathway data</li>
<li>Add short read transcriptomic data to HMP, DIABIMMUNE, (TODO ask Dan for the list)</li>
</ol>
<h2 id="workflow">Workflow</h2>
<p>Generate:</p>
<ul>
<li>taxon counts</li>
<li>gene and pathway abundance, summarized by gene and pathway</li>
<li>gene and pathway abundance, stratified by organism, to enable future extensions to displays</li>
<li>a fraction of reads that didn’t map, and if possible, a separate fraction filtered out as host reads</li>
</ul>
<p>Out of scope:</p>
<ul>
<li>metagenome-assembled-genomes</li>
<li>other deliverables requiring de novo assembly</li>
</ul>
<p>Non-functional requirements:</p>
<ul>
<li>the workflow should  run on the PMACS cluster</li>
<li>the workflow should not use DJob</li>
</ul>
<h2 id="data-model">Data model</h2>
<p>Do whatever is required to get the data to display, but possibly:<br>
-add a column to abundance tables that indicates source of the calculation : 16s/dna</p>
<ul>
<li>new tables for genes and pathways</li>
</ul>
<p>Do not attempt to create new record types before extending MicrobiomeSampleRecord.</p>
<h2 id="sample-record-page">Sample record page</h2>
<p>The sample page should say what data is available for it (16s / dna short read).</p>
<p>For samples containing short read data, there should be information about:</p>
<ul>
<li>taxon counts: separate table for the DNA reads</li>
<li>separate section for functional units</li>
<li>genes, as a flat table: description and abundance</li>
<li>pathways as a larger table: EC number, pathway source, description (as multiple columns), coverage, and abundance</li>
</ul>
<h2 id="searches">Searches</h2>
<p>Search by sample details should let the user specify they only want samples for which there is short read data.</p>
<p>For studies that have any short read data, there should be a new search by pathway abundance, working analogously to the search by taxon abundance.</p>
<p>Out of scope: better display of abundance searches - it is currently not easy to find out e.g. in which studies a certain taxon was found, or what sample details they’re associated with.</p>
<h2 id="analysis-tools">Analysis tools</h2>
<p>Allow viewing the abundance of genes and pathways for groups of samples. Do this by either extending the relative abundance app to toggle between taxa, genes, and pathways, or cloning the relative abundance app and modifying it slightly.<br>
Show the toggle, or the extra app, only when there is data.</p>
<p>Out of scope: other apps.</p>
<h2 id="future-directions">Future directions</h2>
<h3 id="new-displays-for-the-same-data">New displays for the same data</h3>
<p>With some kind of overhaul bringing better display and querying tools for the data generated by the workflow could support more use cases.</p>
<h4 id="displays-of-gene-abundance-stratified-by-organism">Displays of gene abundance stratified by organism</h4>
<p>Has some value for explaining the summary numbers -  drilling into the data to understand where it<br>
This would be useful for viewing an organism across samples. If consistently missing or enriched, it potentially indicates novel strains.</p>
<h4 id="searches-and-record-pages-for-pathways">Searches and record pages for pathways</h4>
<p>Tells the user everything MicrobiomeDB knows about a particular pathway.</p>
<h3 id="similar-displays-and-queries-for-transcriptomic-read-data">Similar displays and queries for transcriptomic read data</h3>
<p>I think there are some for the Human Microbiome Project but I’m not sure how commonly this is done.</p>
<h3 id="other-displays-for-other-data-types">Other displays for other data types</h3>
<p>Currently no ideas from me on this, but there are many research methods out there. Might need discussion.</p>

    </div>
  </div>
</body>

</html>
