<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Expression data in WormBase ParaSite</title>
  <link rel="stylesheet" href="https://stackedit.io/style.css" />
</head>

<body class="stackedit">
  <div class="stackedit__left">
    <div class="stackedit__toc">
      
<ul>
<li><a href="#parasite-expression-requirements-specification">parasite-expression: requirements specification</a>
<ul>
<li><a href="#intention">Intention</a></li>
<li><a href="#data-we-can-offer">Data we can offer</a></li>
<li><a href="#studies-page">Studies page</a></li>
<li><a href="#the-need-for-more-than-just-studies-pages">The need for more than just studies pages</a></li>
<li><a href="#gene-pages">Gene pages</a></li>
<li><a href="#integration-with-jbrowse">Integration with JBrowse</a></li>
<li><a href="#integration-with-biomart">Integration with BioMart</a></li>
<li><a href="#miscellaneous">Miscellaneous</a>
<ul>
<li></li>
</ul>
</li>
</ul>
</li>
</ul>

    </div>
  </div>
  <div class="stackedit__right">
    <div class="stackedit__html">
      <h1 id="parasite-expression-requirements-specification">parasite-expression: requirements specification</h1>
<pre><code>(Wojtek's comments in this format)
As specification, this is intentionally opinionated and worded decisively, but these are all strong opinions held weakly.
Tell me what the document is wrong about and what it is missing!

Expression Atlas had to make choices and address problems in the same domain.
I refer to their choices a lot, especially when I think we should diverge.
</code></pre>
<h2 id="intention">Intention</h2>
<p>Add a scaled down and customised Expression Atlas, in its selected capabilities, to WormBase ParaSite.<br>
We want to achieve:</p>
<ul>
<li>a workflow to import alignment and quantification data from RNASeq-er</li>
<li>a process importing quantification data from RNASeq-er and performing analyses specified by curator</li>
<li>for each study, our site can show <em>all results from this study</em></li>
<li>for each gene, our site can show <em>everything we have for the gene</em></li>
<li>consistency with the rest of WormBase ParaSite
<ul>
<li>link to aligned reads in genome browser</li>
<li>basic querying and bulk export through BioMart</li>
</ul>
</li>
</ul>
<p>In this project, we do not intend to deliver:</p>
<ul>
<li><s>ability to aggregate data for groups of genes</s></li>
<li><s>ability to search by biological condition</s></li>
<li><s>consistency with other EBI resources</s></li>
<li><s>ability to see results broken down by transcript</s></li>
<li><s>good support for organisms with as many studies as C. elegans</s></li>
<li><s>data displayable on the genome browser</s></li>
</ul>
<h2 id="data-we-can-offer">Data we can offer</h2>
<p>We should use the underlying data from RNASeq-er. They offer counts, FPKM, and TPMs from Kallisto and htseq2 quantification programs.</p>
<h3 id="assay-vs-run">Assay vs run</h3>
<pre><code>"run" corresponds to one .bam file
"assay" is a group of runs
"study" is a group of runs too - but in a study, we divide runs into assays
"contrast" is a comparison of two assays
</code></pre>
<p>RNASeq-er analyses runs. Expression Atlas - or us - need to provide metadata so that different runs are grouped into assays as replicates. Insights from the study come from comparing assays with each other.</p>
<h3 id="analyses-available">Analyses available</h3>
<h4 id="normalized-expression">Normalized expression</h4>
<p>We should provide this as TPMs from htseq2, aggregated by assay - choose median value among replicates for each assay.<br>
Expression Atlas also accounts for technical replicates and quantile normalizes the data, but we don’t intend to.</p>
<h4 id="raw-counts">Raw counts</h4>
<p>This is just quantification data (counts per gene) in a convenient file. I am not yet sure what the format of this convenient file should be.</p>
<h4 id="differential-expression">Differential expression</h4>
<p>DESeq2: annotate each sample such that they group into assays. DESeq2 will fit a distribution to the data, and then for each two assays - a <em>contrast</em> it can report how much they differ.<br>
Expression Atlas always has clear contrasts - reference vs. test. We don’t mind designs where there is no clear test and reference, more complicated designs, e.g. multiple strains, multiple drug treatments - quantify the variation only due to strain, and only due to drug treatment.</p>
<h4 id="no-expression-evidence-vs.-expression-found-somewhere">no expression evidence vs. expression found somewhere</h4>
<p>Some studies can only be used as a “yes or no” measure of expression. This is frequently a valuable insight for our data, and we will use it where available.</p>
<h3 id="what-analyses-for-what-studies">What analyses for what studies</h3>
<p>Normalized expression and raw counts should always be available. Differential expression will be available for whatever contrasts are created - and comparing all assays against all assays will be a default and frequently sensible choice.</p>
<p>The Expression Atlas notion of baseline and differential experiments as distinct is not suitable for us. We want to be able to do different analyses where we think is sensible.</p>
<h2 id="studies-page">Studies page</h2>
<p>Current WormBase ParaSite species page should link to a page with a list of RNASeq studies, where available. All studies for a species should appear on the page in some intentional order, possibly as one long list.<br>
We expect at most ten studies per species in the first release of the functionality, and never have more than fifty in the foreseeable future. We will import no more than a fraction of available <em>C. elegans</em> data, and giving it coverage will never be our major focus.</p>
<h3 id="example-studies---s.-mansoni">Example studies - <em>S. mansoni</em></h3>
<p><em>Imagine the below as one long list, listing all S. mansoni studies we processed. For each study it is clear where we took it from, what the authors said it was, and how we curated and analyzed it. More of this information is available when requested. User can browse through and read the descriptions, and either download data as TSV files or view them in a simple webpage, opened in separate browser tab.</em></p>
<pre><code>Simple experiment - two assays, no clear "reference" vs. "test".
Quotes a publication.
</code></pre>
<h4 id="srp071241-comparison-of-gene-expression-between-female-schistosoma-mansoni-heads-and-tails-wang-and-collins-2016">SRP071241: <em>Comparison of gene expression between female Schistosoma mansoni heads and tails</em> (<a href="">Wang and Collins, 2016</a>)</h4>
<h5 id="comparing-by-organism-part-head-vs-tail">comparing by organism part: head vs tail</h5>
<p>adult, female (<a href="">show all</a>)</p>

<table>
<thead>
<tr>
<th><a href="">DESeq2</a></th>
<th><a href="">counts</a></th>
<th><a href="">aggregated expression (TPM)</a></th>
</tr>
</thead>
<tbody></tbody>
</table><br>
<pre><code>Typical differential experiment:
 - something weird done to the animal
 - very specific conditions that facilitated the above
 - expression with and without the weird thing
</code></pre>
<h4 id="srp017443-adult-somatic-stem-cells-in-the-human-parasite-schistosoma-mansoni">SRP017443: <em>Adult somatic stem cells in the human parasite Schistosoma mansoni</em></h4>
<h5 id="comparing-by-irradiation-200gy-vs-none">comparing by irradiation: ‘200Gy’ vs none</h5>
<p>age - Perfused from mice 43 days post-infection, culture conditions - in vitro 48 hours, posterior somatic tissue, strain NMRI, male (<a href="">show all</a>)</p>

<table>
<thead>
<tr>
<th><a href="">DESeq2</a></th>
<th><a href="">counts</a></th>
<th><a href="">aggregated expression (TPM)</a></th>
</tr>
</thead>
<tbody></tbody>
</table><br>
<pre><code>This is a typical baseline experiment:
 - attempt to create "normal" conditions for the animal
 - tries to find genes specific to aspects of the animal's biology
</code></pre>
<h4 id="e-mtab-451--rna-seq-of-schistosoma-mansoni-flatworms-larva-and-adult-individuals-at-different-life-stages">E-MTAB-451:  <em>RNA-Seq of Schistosoma mansoni (flatworms) larva and adult individuals at different life-stages</em></h4>
<h5 id="comparing-by-developmental-stage-cercarium-3hr-schistosomulum-24hr-schistosomulum-platyhelminth-adult">comparing by developmental stage: cercarium, 3hr schistosomulum, 24hr schistosomulum, platyhelminth adult</h5>
<p>whole organism, mixed sex, frozen specimen, strain: Puerto Rican  (<a href="">show all</a>)</p>

<table>
<thead>
<tr>
<th><a href="">DESeq2</a></th>
<th><a href="">counts</a></th>
<th><a href="">aggregated expression (TPM)</a></th>
</tr>
</thead>
<tbody></tbody>
</table><br>
<pre><code>Many assays, and the dataset can be interrogated in many ways.
Challenging for us to support, but it's very important that we are able to.
</code></pre>
<h4 id="erp002113-praziquantel-mode-of-action-and-resistance">ERP002113: <em>Praziquantel mode of action and resistance</em></h4>
<h5 id="comparing-by">comparing by</h5>
<ul>
<li><strong>strain</strong>: LE, LEPZQ, NMRI</li>
<li><strong>developmental stage</strong>: juvenile vs adult</li>
<li><strong>treatment</strong>: Praziquantel vs control</li>
<li><strong>sex</strong>: male, female, mixed</li>
</ul>
<p>whole organism (<a href="">show all</a>)</p>

<table>
<thead>
<tr>
<th><a href="">DESeq2: strain</a></th>
<th><a href="">DESeq2: developmental stage</a></th>
<th><a href="">DESeq2: treatment</a></th>
<th><a href="">DESeq2: sex</a></th>
<th><a href="">counts</a></th>
<th><a href="">aggregated expression (TPM)</a></th>
</tr>
</thead>
<tbody></tbody>
</table><br>
<pre><code>#### Question marks about UI
The following information is available, but I'm not sure whether and how to include:
- data provider
- number of runs
- links to study page
"Show all" intends to lead to experiment design, which would list various properties of the runs in the study and how they're grouped into assays. I don't know how to design this better now.
</code></pre>
<h2 id="the-need-for-more-than-just-studies-pages">The need for more than just studies pages</h2>
<p>It is very simple to provide only the above:</p>
<ul>
<li>data can be provided as files on the FTP server</li>
<li>webpage per species describing the studies and linking to these files - this can even be written and corrected manually</li>
</ul>
<p>We believe listing studies and analyzing them is not valuable enough - the study authors already analyze their data, better than we can do.</p>
<pre><code>## One idea: analyze data on demand
The idea was to offer information about studies, and then let the user interact with it: change how runs are grouped into assays, select analyses to be done, and wait for results.
### Why not
- The user won't know what to select unless we curate the metadata well
- I don't foresee many analysis options, so there is not much value in providing capability to do them dynamically
- It is very valuable for the user to be able choose a gene and then see data for this gene from different studies
### Why it appealed to me for a while
- still quite simple - no need for a database storing individual results
- no capability to search by gene could be excused if users are "digging out" the data themselves
</code></pre>
<h2 id="gene-pages">Gene pages</h2>
<p>Gene pages make this into a much more challenging software project.<br>
With searching by gene the system needs to understand what the studies are, how to extract the data for each gene, and how to present it.<br>
We think it’s very important to let a user see results for a gene of interest across all studies.</p>
<h3 id="data-we-can-offer-per-gene">Data we can offer per gene</h3>
<p>The analyses will yield:</p>
<ul>
<li>expression level for each assay</li>
<li>fold change and some measure of certainty like p-value  for each contrast</li>
</ul>
<p>We will have uniform criteria for when we consider data points significant or not:</p>
<ul>
<li>expression of at least 0.5 TPM in some assay to be significant in the study</li>
<li>p. value &lt; 0.05 and at least two-fold change or similar to be significant in the contrast</li>
</ul>
<p>We will present significant results first.</p>
<pre><code>A notion of attaching significance to results helps with querying BioMart.
I would find it helpful if we discussed it:
 - is it meaningful to for us to say that a result is significant vs. not based on a threshold
 - are the criteria right?
</code></pre>
<h3 id="visualization">Visualization</h3>
<p>We will prioritize clarity, which we will first attempt to achieve with simple data representations like a two column table. We also have the following graphical representations in mind.</p>
<h4 id="expression-per-assay">Expression per assay</h4>
<p>We consider a bar chart: different assays on one axis, expression on another.</p>
<h4 id="contrasts">Contrasts</h4>
<p>This could be represented as a square matrix with assays in both rows and columns, cell representing a contrast, and cell color representing fold change.</p>
<h2 id="integration-with-jbrowse">Integration with JBrowse</h2>
<p>When consuming per-gene information, it is essential that the user is able to verify data by looking at more basic data it was inferred from. We currently present such data in Ensembl’s genome browser and in JBrowse, and we will link to JBrowse.</p>
<p>Specially crafted link for each study will lead from a gene page into a JBrowse display with correct data open. This is worthwhile despite the limitations of JBrowse : no more than 30 runs are allowed, and many fewer runs fit on the screen.</p>
<h2 id="integration-with-biomart">Integration with BioMart</h2>
<p>We believe providing queries based on expression values in BioMart is of very limited value. Motivated users will analyze study results in detail after downloading the data from us, but it is infeasible to allow them to do so through the BioMart interface: complex queries that can be interesting vary per study and BioMart doesn’t allow this.</p>
<p>We only intend to support queries that are not specific to individual studies. The initial release should support these queries:</p>
<ul>
<li>is there expression for gene anywhere?</li>
<li>is there significant expression data for any of chosen studies? (user multi-selects a study)</li>
<li>is there significant enrichment for any of chosen contrasts? (user multi-selects a contrast)</li>
</ul>
<p>We will support export of expression data through BioMart in tabular form.</p>
<h2 id="miscellaneous">Miscellaneous</h2>
<h4 id="how-much-work-will-it-take">How much work will it take?</h4>
<p>We expect at least half a year of active development before getting minimal viable functionality. Many people will be involved in the work and its different aspects.</p>
<h4 id="can-we-analyze-data-between-the-studies-in-a-meta-analysis-and-say-as-wormbase-what-we-think-the-expression-is">Can we analyze data between the studies in a meta-analysis and say as WormBase what we think the expression is?</h4>
<p>Provenance of information is important to us across our resource. Usually we provide a single answer to a question - e.g. we say with confidence where the repeats are in the genome, calculating them with the best tool available to us. We pool available sources and provide a single description for a gene, quoting e.g. UniProt that <code>Smp_144070</code> is a <code>putative gastric mucin</code>.<br>
We believe it is fundamentally impossible to do so with expression data. We can pool runs from different studies and apply normalization techniques (Compare: <em>Lu et al, 2018</em>) but this will create another study among many, perhaps to be featured as most valuable.</p>
<h4 id="is-this-only-for-expression-data">Is this only for expression data?</h4>
<p>The following kind of data will fit:</p>
<ul>
<li>Study captures an organism in a few carefully controlled states</li>
<li>Each of these states generates data per gene</li>
<li>Insights come from comparing per-gene results, to link the difference in experimental conditions with gene function</li>
<li>The user will mostly browse through analyzed data</li>
<li>It is interesting to look through an analysis and see genes that stand out, and to look at results in all analyses for a gene of interest</li>
</ul>
<h4 id="what-limitations-are-we-comfortable-with">What limitations are we comfortable with?</h4>
<p>The solution will be most suited for a small number of studies with not too much analysed data per species. In the proposed design, all species will all be listed on one page. We can have no more than perhaps fifty studies per species if the page is to stay clear, navigable, and load smoothly.</p>
<p>The studies may only have modest amounts of processed data because we do not plan dedicated infrastructure to it. We will budget one megabyte per assay in a core species database, and half a megabyte per assay in BioMart.</p>
<h4 id="is-this-only-for-wormbase-parasite">Is this only for WormBase ParaSite?</h4>
<p>Plants or mosquitoes communities are in a similar position of having available genomes for many species, with modest amount of functional genomics data for each. The work will stay generic enough to e.g. be easily adapted to a new Ensembl website in 2020, and to be adaptable throughout the lifetime of WormBase ParaSite.</p>
<pre><code>I am not sure where else it could currently go apart from WBPS, and who would want it.
I am nevertheless dreaming big.
</code></pre>

    </div>
  </div>
</body>

</html>
