<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Lab meeting 17.04.2018</title>
  <link rel="stylesheet" href="https://stackedit.io/style.css" />
</head>

<body class="stackedit">
  <div class="stackedit__html"><h1 id="wbps10-and-plans-for-wbps11">WBPS10 and plans for WBPS11</h1>
<p>Lab meeting, 17.04.2018</p>
<h2 id="we-have-released-wbps10">We have released WBPS10</h2>
<h3 id="new-genomes-and-updates">New genomes and updates</h3>
<ul>
<li>two Diploscapter species - free-living nematodes, studied in the context of reproduction</li>
<li><em>Ascaridae</em> from U.C. Denver
<ul>
<li>much better assembly of <em>Ascaris suum</em>, and new assembly: <em>Parascaris univalens</em></li>
<li>annotation had errors, we’ll have another update</li>
</ul>
</li>
<li><em>Echinococcus canadensis</em> that was going to be published in July 2017 and wasn’t</li>
<li>minor stuff from Wormbase</li>
</ul>
<h3 id="new-features">New features</h3>
<ul>
<li>None really</li>
</ul>
<h3 id="software-updates">Software updates</h3>
<ul>
<li>Ensembl 89</li>
</ul>
<h3 id="what-each-release-takes">What each release takes</h3>
<h4 id="dependent-on-release-size">Dependent on release size</h4>
<ul>
<li>New genomes and updates</li>
<li>New features</li>
</ul>
<h4 id="constant-overhead">Constant overhead</h4>
<ul>
<li>Dev ops
<ul>
<li>Prepare the environment</li>
<li>Install new software</li>
<li>etc.</li>
</ul>
</li>
<li>All-species pipeline runs
<ul>
<li>Compara</li>
<li>Protein features</li>
<li>BioMart</li>
<li>FTP site</li>
</ul>
</li>
</ul>
<h4 id="should-not-be-needed-next-time">Should not be needed next time</h4>
<ul>
<li>Find my way around the project</li>
<li>Fill holes in documentation</li>
<li>Fix biomart :)</li>
<li>Babysit running jobs so much</li>
</ul>
<h2 id="plans-for-wbps11">Plans for WBPS11</h2>
<h3 id="new-genomes-and-updates-1">New genomes and updates</h3>
<h4 id="wtsi">WTSI</h4>
<h5 id="ready-for-me-to-process">Ready for me to process</h5>
<ul>
<li>Trichuris muris (WS 264)</li>
</ul>
<h5 id="if-theyre-ready-by-mid-may">If they’re ready by mid-May</h5>
<p>Schistosoma mansoni, Haemonchus contortus, Hymenolepis microstoma</p>
<h5 id="more">More?</h5>
<p>Dicrocoelium dendriticum: assembly in NCBI, but no annotation. Why not?</p>
<h4 id="coming-soon">Coming soon</h4>
<h5 id="uc-denver-ascaridae">UC Denver Ascaridae</h5>
<p>Publish fixed gene models (many are currently off by one base due to an error)</p>
<h5 id="liverpool-f.-hepatica">Liverpool F. hepatica</h5>
<p>Steve Patterson is submitting it to an archive now</p>
<h5 id="s.-mediterranea">S. mediterranea</h5>
<p>Planarians are free-living platyhelminths, should be good for our comparative genomics</p>
<h5 id="onchocerca-flexulosus">Onchocerca flexulosus</h5>
<p><a href="https://www.ncbi.nlm.nih.gov/assembly/GCA_002249935.1/">Genome Institute Washington, submitted 2017/08/09</a><br>
There’s already a <a href="http://parasite.wormbase.org/Onchocerca_flexuosa_prjeb512">draft assembly from WTSI</a>, different bioproject<br>
I think we should have them both.</p>
<h5 id="meloydogyne-data-from-inra">Meloydogyne data from INRA</h5>
<h5 id="other">Other</h5>
<ul>
<li>Macrostomum lignano update (free living, for regeneration studies)</li>
<li>Oschieus tipulae (C. elegans satellite)</li>
<li>Maybe some more</li>
</ul>
<h4 id="genomes-without-annotation">Genomes without annotation</h4>
<h5 id="pest-people-that-replied-so-far">Pest people that replied so far</h5>
<p><i>No annotation, these are poor genomes, are you sure you want them?</i></p>
<ul>
<li>ARS USDA</li>
<li>T. Kkuchi, U of Miyazaki</li>
</ul>
<h5 id="other-planarians-from-oxford">Other planarians from Oxford</h5>
<pre><code>I’m afraid there are no annotations at this point for these assemblies.
All the best,
Aziz
</code></pre>
<h4 id="rnaseq-data">RNASeq data</h4>
<p><strong>Current target:</strong> have (almost) all public data available as tracks</p>
<ul>
<li>write an import pipeline from RNASeq-er
<ul>
<li>take processed track files and host them from Sanger
<ul>
<li>TODO: check how much space it will take etc.</li>
</ul>
</li>
<li>create track descriptions
<ul>
<li>semi-manually, sadly</li>
</ul>
</li>
<li>run a JBrowse script to have it included and displayed</li>
</ul>
</li>
<li>work out workflow with RNASeq-er</li>
<li>work out transfer and storage to Sanger</li>
<li>versioning per assembly</li>
</ul>
<p><strong>On the horizon:</strong> quantified data per gene<br>
( not starting this yet )</p>
<h3 id="bsp-ideas-for-new-features">BSP ideas for new features</h3>
<h4 id="regex-filter-on-sequences-through-biomart">Regex filter on sequences through BioMart</h4>
<p>This depends on what BioMart will let me do, this might be technically hard (even a simpler variant, like filtering on a subsequence)</p>
<h4 id="add-comments-to-parasite">Add comments to ParaSite</h4>
<ul>
<li>Location / Format
<ul>
<li>Gene page, left hand side menu</li>
<li>Originally: “Submit a comment”</li>
<li>Becomes: “Comments” when it has some</li>
<li>A page where we explain what comments we invite, and a text box</li>
</ul>
</li>
<li>Use cases
<ul>
<li>Iain has a list of proteins he found something about, has nowhere to publish it, but it’s a good list</li>
</ul>
</li>
<li>Concerns (mine)
<ul>
<li>I would need to moderate the comments and remove impolite / irrelevant content and I don’t really want to</li>
<li>I expect no one will use it</li>
<li>Even if people use it, it will look like nobody does, because there will be nothing said about most genes</li>
</ul>
</li>
</ul>
<h4 id="programmatic-access-to-protein-stats">Programmatic access to protein stats</h4>
<ul>
<li>things like: isoelectric point</li>
<li>displayable on the website, but not accessible programmatically</li>
<li>add them to API or BioMart</li>
<li>doesn’t fit too elegantly in any API section</li>
<li>in general: niche, but easily doable and requested</li>
</ul>
<h3 id="planned-software-updates-and-fixes">Planned software updates and fixes</h3>
<ul>
<li>Ensembl 92</li>
<li>new xref pipeline from Ensembl</li>
<li>New version of JBrowse (we’re on a year old version)</li>
<li>Kevin and Faye found a BioMart bug, I’ll fix it</li>
</ul>
</div>
</body>

</html>
