<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Lab meeting 24.07.2018</title>
  <link rel="stylesheet" href="https://stackedit.io/style.css" />
</head>

<body class="stackedit">
  <div class="stackedit__html"><h3 id="wbps11-status">WBPS11 status</h3>
<h4 id="new-species-and-updates">New species and updates</h4>
<h5 id="largest-updates">Largest updates:</h5>
<pre><code>- S. mansoni and H. contortus, our assemblies  
- T. muris as a new Wormbase core species  
- Fasciola hepatica from Liverpool  
</code></pre>
<h5 id="also">Also:</h5>
<pre><code>- Plectus sambesii  
- Three species of Meloidogyne (INRA)  
- Macrostomum lignano, new assembly  
- Caenorhabditis sp. 34 (T. Kikuchi)  
- Oschieus tipulae  
- Caenorhabditis latens and remanei from Oregon  
- 4 other (annotation updates)  
</code></pre>
<h4 id="all-species-pipelines">All-species pipelines</h4>
<h5 id="reran-for-all-species">Reran for all species:</h5>
<ul>
<li>CMScan (noncoding RNAs): to avoid inconsistency over versions</li>
<li>Cross-references ( we have a better pipeline now, more description of this pipeline to follow)</li>
</ul>
<h5 id="also-as-usual">Also, as usual:</h5>
<ul>
<li>Compara</li>
<li>InterProScan etc.</li>
</ul>
<h4 id="rnaseq-tracks">RNASeq tracks</h4>
<p>(described later)</p>
<h4 id="web-dev-by-tuan">Web dev: by Tuan</h4>
<p>Comments in ParaSite gene pages + more</p>
<h4 id="releasing-wbps-11">Releasing WBPS 11</h4>
<p>No planned date yet, but soon. BioMart build needs to finish.</p>
<h2 id="cross-references-pipeline">Cross-references pipeline</h2>
<p>It connects our gene IDs with IDs from other resources. Desired result:</p>
<p><strong>WBGene00001135</strong><br>
<em>Probable vesicular glutamate transporter eat-4 [Source:UniProtKB/Swiss-Prot;Acc:<a href="%5Bhttp://www.uniprot.org/uniprot/P34644">P34644</a>]</em>](<a href="http://www.uniprot.org/uniprot/P34644">http://www.uniprot.org/uniprot/P34644</a>)%5D*)</p>

<table>
<thead>
<tr>
<th>External database</th>
<th>database identifier</th>
</tr>
</thead>
<tbody>
<tr>
<td>UniGene</td>
<td>Cel.19624</td>
</tr>
<tr>
<td>EntrezGene</td>
<td>Probable vesicular glutamate transporter eat-4</td>
</tr>
</tbody>
</table><h3 id="pipeline-overview">Pipeline overview</h3>
<ul>
<li>Download the sources</li>
<li>Parse the sources and pick relevant parts</li>
<li>Map to our genes</li>
<li>Extract results</li>
</ul>
<h3 id="gains-from-using-the-new-pipeline">Gains from using the new pipeline</h3>
<p>Fresh data for all species. Connected us to more resources (RNACentral). Better gene descriptions. Code/process to keep these benefits in future releases.</p>
<h3 id="kinds-of-references">Kinds of references</h3>
<h4 id="direct-through-ids">Direct (through IDs)</h4>
<p>e.g. RefSeq</p>
<pre><code>LOCUS NP_499023 576 aa linear INV 05-JUN-2018  
DEFINITION Probable vesicular glutamate transporter eat-4 [Caenorhabditis  
elegans].  
ACCESSION NP_499023  
...  
[CDS]([https://www.ncbi.nlm.nih.gov/nuccore/NM_066622.4?from=1&amp;to=1731](https://www.ncbi.nlm.nih.gov/nuccore/NM_066622.4?from=1&amp;to=1731)) 1..576  
...  
/db_xref="WormBase:[WBGene00001135]([https://www.wormbase.org/search/gene/WBGene00001135](https://www.wormbase.org/search/gene/WBGene00001135))"  
</code></pre>
<h4 id="checksum">Checksum</h4>
<p>UniParc, RNACentral</p>
<pre><code>URS0000000001 6bba097c8c39ed9a0fdf02273ee1c79a  
URS0000000002 1fe2f0e3c3a2d6d708ac98e9bfb1d7a8  
URS0000000003 7bb11d0572ff6bb42427ce74450ba564  
...  
</code></pre>
<h4 id="sequence-match">Sequence match</h4>
<p>E.g. UniProt</p>
<pre><code>&gt;sp|P34644  
MSSWNEAWDRGKQMVGEPLAKMTAAAASATGAAPPQQMQEEGNENPMQMHSNKVLQVMEQ  
TWIGKCRKRWLLAILANMGFMISFGIRCNFGAAKTHMYKNYTDPYGKVHMHEFNWTIDEL  
...  
</code></pre>
<h4 id="dependent--transitive">Dependent / transitive</h4>
<p>E.g. RefSeq also mentions WikiGene ids</p>
<h2 id="rnaseq-tracks-1">RNASeq tracks</h2>
<p>We have some RNASeq tracks but not as useful as they could be:<br>
<a href="%5Bhttp://parasite.wormbase.org/jbrowse/index.html?data=%2Fjbrowse-data%2Fschistosoma_mansoni_prjea36577%2Fdata&amp;loc=Smp.Chr_1%3A2559064..2581510&amp;tracks=DNA%2CLow_Complexity_Region_(Dust)%2C010_ERS017679&amp;highlight=%5D(http://parasite.wormbase.org/jbrowse/index.html?data=%2Fjbrowse-data%2Fschistosoma_mansoni_prjea36577%2Fdata&amp;loc=Smp.Chr_1%3A2559064..2581510&amp;tracks=DNA%2CLow_Complexity_Region_(Dust)%2C010_ERS017679&amp;highlight=)">JBrowse installed</a>, and a few chosen tracks aligned with Ensembl Genomes code</p>
<h4 id="rnaseq-er"><a href="%5Bhttps://www.ebi.ac.uk/fg/rnaseq/api/%5D(https://www.ebi.ac.uk/fg/rnaseq/api/)">RNASeq-er</a></h4>
<p>Gets data from ENA, aligns the reads, then does quantification.</p>
<h4 id="plan">Plan</h4>
<ul>
<li>take the reads, provided as BigWig files</li>
<li>serve them from Sanger’s NGS server</li>
<li>get metadata about runs from various places</li>
</ul>
<ul>
<li>ENA: descriptions of studies</li>
<li>RNASeq-er: how well they aligned</li>
<li>ArrayExpress: curated metadata</li>
</ul>
<ul>
<li>assemble metadata into categories</li>
<li>make nice JBrowse displays</li>
<li>serve through EBI FTP server, link from gene pages</li>
</ul>
<h4 id="project-status">Project status</h4>
<p>It’s working :) Not all data has deployed yet, the displays are still rough.</p>
<h4 id="demo">DEMO</h4>
<h4 id="metadata-example">Metadata example</h4>
<pre><code>ERR576952:  
ENA first public: 2014-11-12  
ENA last update: 2016-04-19  
PubMed references: 25887684  
Sample Name: ERS524693  
Study description: 'RNA from eggs, metacercariae, newly emerged juveniles, 21 day juveniles and adults were sequenced in order to examine changes in expression through development. This study was led by Prof John Dalton, University of McGill, Canada.'  
Study title: RNAseq analysis of the liver fluke Fasciola hepatica  
dev stage: adult  
label: 'ERR576952: adult'  
urlTemplate: [http://ngs.sanger.ac.uk/production/parasites/wormbase/RNASeq_alignments/fasciola_hepatica_prjeb25283/ERR576952.bw](http://ngs.sanger.ac.uk/production/parasites/wormbase/RNASeq_alignments/fasciola_hepatica_prjeb25283/ERR576952.bw)  
SRR4039330:  
ENA first public: 2017-09-25  
ENA last update: 2017-09-25  
Organism: Fasciola hepatica  
Sample Name: 'TCBZ - ABZ susceptible Fasciola hepatica'  
Sex: hermaphrodite  
Study title: Transcriptomic analysis of F. hepatica Southamerican drug resistant isolates  
biomaterial provider: Hugo Solana. Universidad de Tandil  
colection date: 2014  
development stage: Adult  
geographic location: Mexico  
host: Ovis aries  
isolate: Cenapa  
label: 'SRR4039330: TCBZ - ABZ susceptible Fasciola hepatica, Cenapa'  
model: Invertebrate  
organism part: Whole worm  
package: Invertebrate.1.0  
synonym: Cenapa1  
urlTemplate: [http://ngs.sanger.ac.uk/production/parasites/wormbase/RNASeq_alignments/fasciola_hepatica_prjeb25283/SRR4039330.bw](http://ngs.sanger.ac.uk/production/parasites/wormbase/RNASeq_alignments/fasciola_hepatica_prjeb25283/SRR4039330.bw)  
  
</code></pre>
<h4 id="factors">Factors</h4>
<p>“Characteristics essential to difference between the runs”. I want to identify these factors to make good JBrowse track selectors.</p>
<h5 id="current-approach">Current approach</h5>
<ul>
<li>Take characteristic types that change value</li>
<li>Curate manually :(</li>
</ul>
<h5 id="example-f.-hepatica-studies">Example: f. hepatica studies</h5>
<pre><code>---  
# - Drug resistance  
- Sample Name  
# - biomaterial provider  
- dev stage  
# - geographic location  
# - host  
- isolate  
# - synonym  
</code></pre>
<p>Some ideas:</p>
<ul>
<li>maintain a list of characteristics that probably aren’t going to be good for no species, like <code>synonym</code></li>
<li>rank the types somehow</li>
<li>?</li>
</ul>
</div>
</body>

</html>
